compiler/semantics.cpp

```diff
@@ -86,6 +86,26 @@ static inline OpFunc TokenToOpFunc(int token) {
     }
 }
 
+static void
+CheckFunction(const value* lval)
```

Comment:
Not to be stingy, but for others reading this code `CheckFunction` leaves me asking "for what?"

Reply:
If I knew the answer I'd tell you!

Reply:
Okay, I see. It's serving double-duty, it's supposed to also check whether an expression is using a value but not returning one. In that case, I think we can eliminate it.